### DISCUSSION TITLE
Trying to create a custom mapping for terminal

### DISCUSSION THREAD

**Discussion by BrimVeyn** [NvChad/NvChad#2431] [Q&A] [answered]
Hi,
I'm trying to create a key bind that would exit terminal mode and go to next window. I'm working with C so I'd like to be able to compile, see eventual compilation errors and go back editing directly. Here's what I came up with and it doesn't work at all. I've been trying simpler mappings in the terminal scope and none worked.
Since I'm new to lua, I'm surely missing something so here I am :)

```
M.custom = {
  i = {
     ["jk"] = { "<ESC>", "escape insert mode" , opts = { nowait = true }},
    -- ...
  },
  t = {
    ["<C-.>"] = { "<C-x><C-w>w", "Escape terminal mode and go back to file editing" },
  },
}
```

**Comment by siduck** [comment, marked as answer]
@BrimVeyn try C-x , we have mapped it by default, please check core/mappings.lua

**Reply by BrimVeyn** [reply]
Thank you for your reply.
I've checked core/mappings.lua and seen the mapping and understood it, no problem.

I wanted to create an other bind that would execute C-x plus C-ww since thats what you have to do to get back to the file you're editing.
Is it possible do to so ?

**Reply by BrimVeyn** [reply]
```
  t = {
    ["<C-z>"] = { vim.api.nvim_replace_termcodes("<C-\\><C-N><C-w>w", true, true, true), "Escape terminal mode" },
  },

```
Finally got the answer by myself after testing multiple shit. Seems like you can't call C-x directly, so I copied the core/mapping.lua line and added what I needed. It works !
